themes/src/main/resources/theme/base/login/template.ftl

```diff
@@ -32,62 +32,55 @@
 </head>
 
 <body class="${properties.kcBodyClass!}">
-    <div id="kc-logo"><a href="${properties.kcLogoLink!'#'}"><div id="kc-logo-wrapper"></div></a></div>
-
-    <div id="kc-container" class="${properties.kcContainerClass!}">
-        <div id="kc-container-wrapper" class="${properties.kcContainerWrapperClass!}">
-
-            <div id="kc-header" class="${properties.kcHeaderClass!}">
-                <div id="kc-header-wrapper" class="${properties.kcHeaderWrapperClass!}"><#nested "header"></div>
-            </div>
-
-            <#if realm.internationalizationEnabled>
-                <div id="kc-locale" class="${properties.kcLocaleClass!}">
-                    <div id="kc-locale-wrapper" class="${properties.kcLocaleWrapperClass!}">
-                        <div class="kc-dropdown" id="kc-locale-dropdown">
-                            <a href="#" id="kc-current-locale-link">${locale.current}</a>
-                            <ul>
-                                <#list locale.supported as l>
-                                    <li class="kc-dropdown-item"><a href="${l.url}">${l.label}</a></li>
-                                </#list>
-                            </ul>
-                        </div>
+  <div class="${properties.kcLoginClass!}">
+    <header class="${properties.kcHeaderClass!}">
+      <div id="kc-logo"><a href="${properties.kcLogoLink!'#'}"><div id="kc-logo-wrapper"></div></a></div>
+    </header>
+    <div class="${properties.kcFormCardClass!} <#if realm.password && social.providers??>${properties.kcFormCardAccountClass!}</#if>">
```

Comment:
This is causing a problem on some pages as "realm.password" and "social.providers" are not available on all pages. I don't quite understand what this is doing and why it adds this class like this. Can you explain?

Reply:
Sure, @stianst. The logical of this piece of code is the login box width is different, when no social provide, the width is 500px. and when adding the social provider the width of login box is 840px.
As here, I need a class to control the width.

Reply:
https://github.com/junezhang/keycloak/pull/1